signature/src/image/mod.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use oci_distribution::Reference;
+
+pub mod digest;
+
+use digest::Digest;
+
+#[derive(EnumString, Display, Debug, PartialEq)]
+pub enum TransportName {
+    #[strum(to_string = "docker")]
+    Docker,
+    #[strum(to_string = "dir")]
+    Dir
+}
+
+// Image contains information about the image which may be used in signature verification.
+pub struct Image {
+    pub reference: Reference,
+    // digest format: "digest-algo:digest-value"
+    pub manifest_digest: Digest,
+
+    // Try signature extensions in registry before using the sigstore.
+    // (https://github.com/containers/image/issues/384)
+    //
+    // FIXME: now don't support, always set to false.
+    // 
+    // issue: https://github.com/confidential-containers/image-rs/issues/12
+    pub registry_xrss_api_support: bool,
+}
+
+impl Image {
+    pub fn default_with_reference(image_ref: Reference) -> Self {
+        Image {
+            reference: image_ref,
+            manifest_digest: Digest::default(),
+            registry_xrss_api_support: false,
+        }
+    }
+
+    pub fn transport_name(&self) -> String {
+        // FIXME: Now only support "docker" transport (and it is hardcoded).
+        // TODO: support "dir" transport.
+        // 
+        // issue: https://github.com/confidential-containers/image-rs/issues/11
+        TransportName::Docker.to_string()
+    }
+
+    pub fn set_manifest_digest(&mut self, digest: &str) -> Result<()> {
+        self.manifest_digest = Digest::try_from(digest)?;
+    }
+}
+
+// Get repository full name:
+// `registry-name/repository-name`
+pub fn get_image_repository_full_name(image_ref: &Reference) -> String {
```

Comment:
wouldn't be easier to split by ':' then get the first element?

Reply:
Nit: I am not sure about `repository_full_name` in the name `get_image_repository_full_name`. The `Reference` encapsulates a string formatted as `registry/repository(:[tag|disgest])?`. And what you consider `repository` is a different meaning in `repository` in `Reference`. Does OCI gives a proper name of the `registry/repository` segment?

Reply:
This function implements the `[full_name](https://github.com/krustlet/oci-distribution/blob/main/src/reference.rs#L137)` function in the `oci_distribution::Reference` library because the `full_name` function is private. OCI standards call both `busybox` and `docker.io/library/busybox` segments "repository". The former is a relative path and the latter is a full path.

> wouldn't be easier to split by ':' then get the first element?

Because ':' may also appear in namespace and some custom tags